src/client/datascience/kernel-launcher/kernelEnvVarsService.ts

```diff
@@ -27,66 +23,60 @@ export class KernelEnvironmentVariablesService {
      * The first argument is an executable, and it is not in the current path.
      * However, when activating the conda env, the path variables are updated to set path to the location where the java executable is located.
      */
-    public async getEnvironmentVariables(resource: Resource, kernelSpec: IJupyterKernelSpec) {
-        const customEditVarsPromise = this.customEndVars.getCustomEnvironmentVariables(resource);
-        customEditVarsPromise.catch(noop); // If we return early, we don't want failing promise go unhandled.
+    public async getEnvironmentVariables(
+        resource: Resource,
+        interpreter: PythonEnvironment | undefined,
+        kernelSpec: IJupyterKernelSpec
+    ) {
         let kernelEnv = kernelSpec.env && Object.keys(kernelSpec.env).length > 0 ? kernelSpec.env : undefined;
         if (!kernelSpec.interpreterPath) {
             traceInfo(
                 `No custom variables for Kernel as interpreter path is not defined for kernel ${kernelSpec.display_name}`
             );
             return kernelEnv;
         }
-        const interpreter = await this.interpreterService
```

Comment:
This is required for non-python environments like Java & the like that belong to conda environments.
Basically we need to actiavte the conda environment & then run the non-python kernels, tha'ts what the error message in line 43 means.
We really need to fix/re-enable the java kernel test on CI, that would have caught this

Reply:
@DonJayamanne Sounds good, I can update this. Also reenabling the Java test seems like a great candidate for debt week. I don't see an explicitly disabled test for that in non-python kernels. Maybe we need to add one? Does the beakerX java kernel run from a conda env?

Reply:
> citly disabled test for that in non-python kernels. Maybe we need to add one? Does the beakerX java kernel run from a conda env?

I thought we had a beakerx java kernel in our tests. I'm certain we have it.

Reply:
We had one. It just opens the notebook though. I don't see a disabled test to actually run it. I'll add an issue for this test.

Reply:
Added #6544